pkg/adapter/runtime.go

```diff
@@ -337,3 +337,40 @@ func IsImageNotFound(err error) bool {
 func (r *LocalRuntime) HealthCheck(c *cliconfig.HealthCheckValues) (libpod.HealthCheckStatus, error) {
 	return r.Runtime.HealthCheck(c.InputArgs[0])
 }
+
+// JoinOrCreateRootlessPod joins the specified pod if it is running or it creates a new user namespace
+// if the pod is stopped
+func (r *LocalRuntime) JoinOrCreateRootlessPod(pod *Pod) (bool, int, error) {
+	if os.Geteuid() == 0 {
+		return false, 0, nil
+	}
+
+	inspect, err := pod.Inspect()
+	if err != nil {
+		return false, 0, err
+	}
+	for _, ctr := range inspect.Containers {
```

Comment:
Nit: this is an optimization, but we could also check if pod shares namespaces (pod.SharesNamespaces() ), and then if so find the infra container (pod.InfraContainerID() ). 
the infra container must be running if any container is running, so we could get the infra pid file if it is running, and go to the return statement at the bottom if not. This would skip the iterative container check, which needs to touch the db for state for each container and could get expensive. Not blocking, and may be too clunky of code to be worth it (as what is here is certainly prettier). Unfortunately, we'd still need the iterative code in the case there isn't an infra container
WDYT
cc: @mheon